tools/topology/topology2/cavs/cavs-rt5682.conf

```diff
@@ -0,0 +1,240 @@
+	PLATFORM 			"none"
+}
+
+# override defaults with platform-specific config
+IncludeByKey.PLATFORM {
+	"mtl"	"platform/intel/mtl.conf"
+}
+
+# include DMIC config if needed.
+IncludeByKey.NUM_DMICS {
+	"[1-4]"	"platform/intel/dmic-generic.conf"
+}
+
+# include HDMI config if needed.
+IncludeByKey.NUM_HDMIS {
+"[3-4]" "platform/intel/hdmi-generic.conf"
+}
+
+#
+# List of all DAIs
+#
+#SSP Index: 0, Direction: duplex
+Object.Dai {
+	SSP."0" {
+		id 		0
+		dai_index	0
+		direction	"duplex"
+		name		SSP0-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP0" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+	SSP."1" {
+		id 		$SPK_ID
+		dai_index	1
+		direction	"playback"
+		name		SSP1-Codec
+		default_hw_conf_id	0
+		sample_bits		32
+		io_clk		$MCLK
+
+		Object.Base.hw_config."SSP1" {
+			id	0
+			mclk_freq	$MCLK
+			bclk_freq	3072000
+			tdm_slot_width	32
+		}
+	}
+}
+
+#
+# Pipeline definitions
+#
+
+# Pipeline ID:1 PCM ID: 0
+Object.Pipeline {
+	# playback pipelines
+	passthrough-playback."1" {
+		index	1
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp0"
+		Object.Widget.copier.1.stream_name	"SSP0 Playback"
+	}
+
+	passthrough-be.1 {
+		index	2
+		direction	playback
+
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp0"
+
+		Object.Widget.copier.1 {
+			dai_index 0
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "SSP0-Codec"
+			node_type $I2S_LINK_OUTPUT_CLASS
+		}
+	}
+
+	passthrough-playback.2 {
+		index	3
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp1"
+		Object.Widget.copier.1.stream_name	"SSP1 Playback"
+	}
+
+	passthrough-be.2 {
+		index	4
+		direction	playback
+		Object.Widget.pipeline.1.stream_name	"passthrough-playback-ssp1"
+
+		Object.Widget.copier.1 {
+			dai_index 1
+			dai_type "SSP"
+			copier_type "SSP"
+			stream_name "SSP1-Codec"
+			node_type $I2S_LINK_OUTPUT_CLASS
+		}
+	}
+
+	# capture pipelines
+	passthrough-capture."1" {
+		index 	7
+		Object.Widget.pipeline.1.stream_name	"passthrough-capture-ssp0"
+		Object.Widget.copier.1.stream_name	"SSP0 Capture"
+	}
+
+	passthrough-be.4 {
+		index		8
+		direction	capture
+		Object.Widget.pipeline.1.stream_name	"passthrough-capture-ssp0"
+
+		Object.Widget.copier."1" {
+			dai_index	0
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	"SSP0-Codec"
+			node_type	$I2S_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+	}
+}
+
+Object.PCM {
+	pcm."0" {
+		name	"Headset"
+		id 0
+		direction	"duplex"
+		Object.Base.fe_dai."Headset" {}
+
+		Object.PCM.pcm_caps."playback" {
+			name "SSP0 Playback"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+
+		Object.PCM.pcm_caps."capture" {
+			name "SSP0 Capture"
+			formats 'S16_LE,S24_LE,S32_LE'
+		}
+	}
+
+	pcm."1" {
+		name	"dummy-spk"
```

Comment:
That doesn't seem like a great name.

What's wrong with 'Speakers'?

Reply:
Ack, MX98357A goes by "dummy speaker" cause it does not requires I2C programming :)